front/components/triggers/CreateWebhookSourceForm.tsx

```diff
@@ -89,75 +91,107 @@ export function CreateWebhookSourceFormContent({
-                              "subscribedEvents",
-                              selectedEvents.filter((e) => e !== event.value),
-                              { shouldValidate: true, shouldDirty: true }
-                            );
-                          }
-                        }}
+          render={({ fieldState }) => {
+            const allEvents = WEBHOOK_PRESETS[provider].events;
+            const allSelected = selectedEvents.length === allEvents.length;
+            const dropDownLabel =
+              selectedEvents.length === 0
+                ? "Select events"
+                : allSelected
+                  ? `All events (${selectedEvents.length}) selected`
+                  : `${selectedEvents.length} event${selectedEvents.length > 1 ? "s" : ""} selected`;
+
+            const handleSelectAll = () => {
+              form.setValue(
+                "subscribedEvents",
+                allEvents.map((e) => e.value),
+                { shouldValidate: true, shouldDirty: true }
+              );
+            };
+
+            const handleUnselectAll = () => {
+              form.setValue("subscribedEvents", [], {
+                shouldValidate: true,
+                shouldDirty: true,
+              });
+            };
+
+            return (
+              <div className="flex flex-col gap-2">
+                <Label htmlFor="subscribedEvents">Subscribed events</Label>
+                <div>
+                  <DropdownMenu>
+                    <DropdownMenuTrigger asChild>
+                      <Button
+                        label={dropDownLabel}
+                        variant="outline"
+                        icon={ChevronDownIcon}
                       />
-                      <div className="grid gap-1.5 leading-none">
-                        <label
-                          htmlFor={`${provider}-event-${event.value}`}
-                          className="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
-                        >
-                          {event.name}
-                        </label>
+                    </DropdownMenuTrigger>
+                    <DropdownMenuContent className="w-72" align="start">
+                      <div className="flex gap-2 p-2">
+                        <Button
+                          label="Select all"
+                          icon={PlusCircleIcon}
+                          variant="primary"
+                          size="xs"
+                          onClick={handleSelectAll}
+                          disabled={allSelected}
+                        />
+                        <Button
+                          label="Unselect all"
+                          icon={XCircleIcon}
+                          variant="primary"
+                          size="xs"
+                          onClick={handleUnselectAll}
+                          disabled={selectedEvents.length === 0}
+                        />
```

Comment:
let's wait on ed's input on those buttons (ping him on slack if needed, in hootl or design, with a screen), but implem lgtm